semantics/executable-spec/src/Data/MemoBytes.hs

```diff
@@ -37,18 +40,24 @@ import Cardano.Binary
   )
 import Data.ByteString.Short (ShortByteString, fromShort, toShort)
 import qualified Data.ByteString.Lazy as Lazy
-import Data.ByteString.Lazy (toStrict)
+import Data.ByteString.Lazy (toStrict,fromStrict)
 import Data.Typeable
 import Data.Coders(runE, Encode, encode,)
 import Codec.CBOR.Write (toLazyByteString)
 import GHC.Generics (Generic)
-import NoThunks.Class (AllowThunksIn (..), NoThunks (..))
+import NoThunks.Class (NoThunks (..))
 import Prelude hiding (span)
+import GHC.TypeLits(Symbol)
+import GHC.Records
+import Codec.CBOR.Read(DeserialiseFailure,deserialiseFromBytes)
 
 -- ========================================================================
 
-data MemoBytes t = Memo {memotype :: !t, memobytes :: {-# UNPACK #-} !ShortByteString}
-  deriving (NoThunks) via AllowThunksIn '["memobytes"] (MemoBytes t)
+data MemoBytes t = Memo !t {-# UNPACK #-} !ShortByteString
```

Comment:
Wait, why the bang on the `ShortByteString` if the goal is to allow thunks in the bytes? The bang and the `UNPACK` pragma directly contradict the goal.

Reply:
At one point we needed the MemoBytes type to not have any fileds of its own. So we changed this. That requirement is no longer needed, we also fixed the contradiction between the ! and the allowing thunks there, and reverted to the old style. Thanks for the advice and the link to the video.